src/main/scala/beam/utils/RespositioingValidation.scala

```diff
@@ -0,0 +1,98 @@
+package beam.utils
+
+import beam.agentsim.events.PathTraversalEvent
+import org.matsim.api.core.v01.events.Event
+import org.matsim.core.events.handler.BasicEventHandler
+import org.matsim.core.events.{EventsUtils, MatsimEventsReader}
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+case class RepositioningDuration(repositioningDurationStart: Double, repositioningDurationEnd: Double)
+
+object RepositioningValidation {
+
+  val vehicleRepositioning: mutable.Map[String, ListBuffer[RepositioningDuration]] =
+    mutable.Map[String, ListBuffer[RepositioningDuration]]()
+
+  def main(args: Array[String]): Unit = {
+    var eventsFileFormat = "xml"
+    if (args.length == 2) {
+      eventsFileFormat = args(1)
+    }
+    if (eventsFileFormat == "csv") {
+      val (eventCount, foundEvent) = readCSVEvents(args(0))
+      println(s"Total event count $eventCount")
+      println(s"Event found $foundEvent")
+
+    } else {
+      val (eventCount, foundEvent) = readXMLEvents(args(0))
+      println(s"Total event count $eventCount")
+      println(s"Event found $foundEvent")
+    }
+    //readEvents("/home/rajnikant/IdeaProjects/beam/output/sf-light/sf-light-1k-xml__2019-10-02_03-28-09/ITERS/it.0/0.events.xml.gz")
+
+  }
+
+  private def readCSVEvents(path: String): (Int, Int) = {
+    val (eventSeq, _) = EventReader.fromCsvFile(path, _ => true)
+    var eventCount = 0
+    var foundEvent = 0
+    eventSeq.foreach(event => {
+      if (event.getEventType == "PathTraversal") {
+        eventCount += 1
+        if (processForRepositioningDebug(event))
```

Comment:
Why does this one not `PathTraversalEvent.apply`?

Reply:
Make the changes as per above comment